### DISCUSSION TITLE
Clarification about origin allow list

### DISCUSSION THREAD

**Discussion by zachfedor** [WalletConnect/walletconnect-monorepo#3596] [Developer Support]
According to the help text when adding origins to the allow list, the following are possible values:

```
<scheme>://<hostname>
<scheme>://<hostname>:<port>
<scheme>://*.<hostname>
```

Does that mean the following is not valid?

```
<scheme>://<substring>*.<hostname>
```

If not, do you have any recommendations on how to handle this situation?

To explain the use case, we use Vercel for test deploys. We'd like to allow our QA team to test Wallet Connect features by opening up our project ID to be used on Vercel domains _that we control_.  A deploy to our project might look like `https://our-organization-SOME-UNIQUE-BUILD-ID.vercel.com`, so an allow list value of `https://our-organization-*.vercel.com` would work perfectly for this situation without allowing the entire `vercel.com` domain.

**Comment by zachfedor** [comment]
Two weeks... Any answers?

**Comment by SergeyVolynkin** [comment]
Even if it is indeed suggested in the https://cloud.walletconnect.com/app UI that you can add `https://*.domain.com` to the allow-list –– such a pattern will not work, you would need to specify a list of full URLs to be allow-listed @zachfedor 
(update: The thing that I was not clear on, is that `https://*.domain.com` did not work for me for URLs that have multiple nesting levels `https://a.b.c.domain.com`)

**Comment by zachfedor** [comment]
> Even if it is indeed suggested

It is, [right here](https://docs.walletconnect.com/2.0/cloud/relay#allowlist). If you didn't bother reading the docs, then how am I supposed to take your answer seriously?

> such a pattern will not work

It does. We were having invalid URL console errors on a certain domain when I tried to use the `https://substring-*.domain.com` pattern, but they disappeared when I dropped the substring and just used `https://*.domain.com`. This is a temporary solution for us, we'd prefer to be more specific about which subdomains to allow, hence my original question.

**Comment by SergeyVolynkin** [comment]
The thing that I was not clear on, is that `https://*.domain.com` did not work for me for URLs that have multiple nesting levels `https://a.b.c.domain.com`

**Reply by zachfedor** [reply]
Interesting, I hadn't tried another level of subdomains. Hopefully they can add some finer-grained control here for both of our sakes.
